src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -612,15 +616,24 @@ def build(self, notification_uuid: str | None = None) -> SlackBlock | SlackAttac
         # build title block
         title_text = f"<{title_link}|*{escape_slack_text(title)}*>"
 
+        has_improvements_feature_flag = features.has(
+            "organizations:slack-block-kit-improvements", self.group.project.organization
+        )
         if self.group.issue_category == GroupCategory.ERROR:
             level_text = None
             for k, v in LOG_LEVELS_MAP.items():
                 if self.group.level == v:
                     level_text = k
 
-            title_emoji = LEVEL_TO_EMOJI.get(level_text)
+            if has_improvements_feature_flag:
+                title_emoji = LEVEL_TO_EMOJI_V2.get(level_text)
+            else:
+                title_emoji = LEVEL_TO_EMOJI.get(level_text)
         else:
-            title_emoji = CATEGORY_TO_EMOJI.get(self.group.issue_category)
+            if has_improvements_feature_flag:
+                title_emoji = CATEGORY_TO_EMOJI_V2.get(level_text)
+            else:
+                title_emoji = CATEGORY_TO_EMOJI.get(self.group.issue_category)
```

Comment:
This might be a good candidate for a function!
```
def _get_title_emoji(self) -> str:
    # check flag here
    # check self.group.issue_category here
    # all future logic ...
    return emoji
```